code/game/objects/structures/oxygen.dm

```diff
@@ -0,0 +1,157 @@
+			playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You unsecure [name].</span>")
+			new /obj/item/wallframe/oxygen_cabinet(loc)
+			qdel(src)
+		return
+
+	if(isrobot(user) || isalien(user))			// No non-humans allowed!
+		return
+
+	if( (istype(I, /obj/item/weapon/tank/internals/emergency_oxygen)) || (istype(I, /obj/item/clothing/mask/breath)) )
+		if(istype(I, /obj/item/weapon/tank/internals/emergency_oxygen))			// If standard oxygen tank in hand, put in cabinet
+			if(!has_oxygen && opened)
+				if(!user.drop_item())
+					return
+				contents += I
+				has_oxygen = I
+				to_chat(user, "<span class='notice'>You place [I] in [src].</span>")
+			else
+				opened = !opened
+
+		else			// If breath mask in hand, put in cabinet
+			if(!has_mask && opened)
+				if(!user.drop_item())
+					return
+				contents += I
+				has_mask = I
+				to_chat(user, "<span class='notice'>You place [I] in [src].</span>")
+			else
+				opened = !opened
+
+	else			// If other items, open/close cabinet
+		opened = !opened
+
+	update_icon()
+
+
+/obj/structure/oxygen_cabinet/attack_hand(mob/user)
+	if(isrobot(user) || isalien(user))			// No non-humans allowed!
+		return
+
+	if(has_oxygen || has_mask)			// If there is either an oxygen tank or breath mask inside cabinet, take one. Priority tank over mask
+		if(has_oxygen)
+			user.put_in_hands(has_oxygen)
+			to_chat(user, "<span class='notice'>You take [has_oxygen] from [src].</span>")
+			has_oxygen = null
+			opened = 1
+		else
+			user.put_in_hands(has_mask)
+			to_chat(user, "<span class='notice'>You take [has_mask] from [src].</span>")
+			has_mask = null
+			opened = 1
+	else
+		opened = !opened
+
+	update_icon()
+
+
+/obj/structure/oxygen_cabinet/attack_tk(mob/user)
+	if(has_oxygen || has_mask)			// Same as before, just with telekinesis
+		if(has_oxygen)
+			has_oxygen.loc = loc
+			to_chat(user, "<span class='notice'>You telekinetically remove [has_oxygen] from [src].</span>")
+			has_oxygen = null
+			opened = 1
+		else
+			has_mask.loc = loc
+			to_chat(user, "<span class='notice'>You telekinetically remove [has_mask] from [src].</span>")
+			has_mask = null
+			opened = 1
+	else
+		opened = !opened
+
+	update_icon()
+
+
+/obj/structure/oxygen_cabinet/attack_paw(mob/user)			// Monkey paws count as hands
+	attack_hand(user)
+	return
+
+
+/obj/structure/oxygen_cabinet/AltClick(mob/living/user)
+	if(user.incapacitated() || !Adjacent(user) || !istype(user))
+		return
+	opened = !opened
+	update_icon()
+
+
+/obj/structure/oxygen_cabinet/update_icon()			// Updates sprite based on contents. Very basic coding. hOw dO I BitFlAG ?!
+	if(!opened)
+		icon_state = "oxygen_closed"
+		return
+	else
+		if(has_oxygen && has_mask)
+			icon_state = "oxygen_full"
+		else
+			if(has_oxygen && !has_mask)
+				icon_state = "oxygen_tank"
+			else
+				if(!has_oxygen && has_mask)
+					icon_state = "oxygen_mask"
+				else
+					icon_state = "oxygen_empty"

```

Comment:
if you name the icon states oxygen-1-1 oxygen-1-0 oxygen-0-1 and oxygen-0-0 respectively you can skip the else if chain and just do icon_state = oxygen-[has_oxygen]-[has_mask]